README.markdown
#SiteLeaf Theme by Kevin Glover

This theme is written using the [Liquid templating lanugage.](https://github.com/Shopify/liquid) 

##Purpose
I developed this theme as an introduction to Liquid and for my personal site 
[glover.io.](http://glover.io) I wanted to create a focused site for 
blogging and sharing my ideas.

##Deploying 
I am using SiteLeaf for my CMS and deploy, but you could use any Jekyll site 
complier as well.

##Google Analytics
To use Google Analytics with this theme, create a subpage under 
Index titled "Google Analytics." Paste the analytics snippet into the body 
of this new page. This code will be inserted on every page. 


CHANGELOG.markdown
# Change log for Siteleaf theme

## Release 1.1
* Added 404 page
* Added changelog!

## Release 1.0
* Initial release of theme
* Homepage template
* Blog page template
* Blog posts template
* Blog post API for pulling posts dynamically


js/menu.js
;(function(){
  /**
 * borderMenu.js v1.0.0
 * http://www.codrops.com
 *
 * Licensed under the MIT license.
 * http://www.opensource.org/licenses/mit-license.php
 * 
 * Copyright 2013, Codrops
 * http://www.codrops.com
 */
 
'use strict';
 
  // http://stackoverflow.com/a/11381730/989439
    function mobilecheck() {
      var check = false;
      (function(a){if(/(android|ipad|playbook|silk|bb\d+|meego).+mobile|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|netfront|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|series(4|6)0|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino/i.test(a)||/1207|6310|6590|3gso|4thp|50[1-6]i|770s|802s|a wa|abac|ac(er|oo|s\-)|ai(ko|rn)|al(av|ca|co)|amoi|an(ex|ny|yw)|aptu|ar(ch|go)|as(te|us)|attw|au(di|\-m|r |s )|avan|be(ck|ll|nq)|bi(lb|rd)|bl(ac|az)|br(e|v)w|bumb|bw\-(n|u)|c55\/|capi|ccwa|cdm\-|cell|chtm|cldc|cmd\-|co(mp|nd)|craw|da(it|ll|ng)|dbte|dc\-s|devi|dica|dmob|do(c|p)o|ds(12|\-d)|el(49|ai)|em(l2|ul)|er(ic|k0)|esl8|ez([4-7]0|os|wa|ze)|fetc|fly(\-|_)|g1 u|g560|gene|gf\-5|g\-mo|go(\.w|od)|gr(ad|un)|haie|hcit|hd\-(m|p|t)|hei\-|hi(pt|ta)|hp( i|ip)|hs\-c|ht(c(\-| |_|a|g|p|s|t)|tp)|hu(aw|tc)|i\-(20|go|ma)|i230|iac( |\-|\/)|ibro|idea|ig01|ikom|im1k|inno|ipaq|iris|ja(t|v)a|jbro|jemu|jigs|kddi|keji|kgt( |\/)|klon|kpt |kwc\-|kyo(c|k)|le(no|xi)|lg( g|\/(k|l|u)|50|54|\-[a-w])|libw|lynx|m1\-w|m3ga|m50\/|ma(te|ui|xo)|mc(01|21|ca)|m\-cr|me(rc|ri)|mi(o8|oa|ts)|mmef|mo(01|02|bi|de|do|t(\-| |o|v)|zz)|mt(50|p1|v )|mwbp|mywa|n10[0-2]|n20[2-3]|n30(0|2)|n50(0|2|5)|n7(0(0|1)|10)|ne((c|m)\-|on|tf|wf|wg|wt)|nok(6|i)|nzph|o2im|op(ti|wv)|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|\-([1-8]|c))|phil|pire|pl(ay|uc)|pn\-2|po(ck|rt|se)|prox|psio|pt\-g|qa\-a|qc(07|12|21|32|60|\-[2-7]|i\-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55\/|sa(ge|ma|mm|ms|ny|va)|sc(01|h\-|oo|p\-)|sdk\/|se(c(\-|0|1)|47|mc|nd|ri)|sgh\-|shar|sie(\-|m)|sk\-0|sl(45|id)|sm(al|ar|b3|it|t5)|so(ft|ny)|sp(01|h\-|v\-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl\-|tdg\-|tel(i|m)|tim\-|t\-mo|to(pl|sh)|ts(70|m\-|m3|m5)|tx\-9|up(\.b|g1|si)|utst|v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|\-v)|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(\-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|yas\-|your|zeto|zte\-/i.test(a.substr(0,4)))check = true})(navigator.userAgent||navigator.vendor||window.opera);
      return check;
    }

    function init() {

      var menu = document.getElementById( 'bt-menu' ),
        trigger = menu.querySelector( 'a.bt-menu-trigger' ),
        // triggerPlay only for demo 6
        triggerPlay = document.querySelector( 'a.bt-menu-trigger-out' ),
        // event type (if mobile use touch events)
        eventtype = mobilecheck() ? 'touchstart' : 'click',
        resetMenu = function() {
          classie.remove( menu, 'bt-menu-open' );
          classie.add( menu, 'bt-menu-close' );
        },
        closeClickFn = function( ev ) {
          resetMenu();
          overlay.removeEventListener( eventtype, closeClickFn );
        };

      var overlay = document.createElement('div');
      overlay.className = 'bt-overlay';
      menu.appendChild( overlay );

      trigger.addEventListener( eventtype, function( ev ) {
        ev.stopPropagation();
        ev.preventDefault();
        
        if( classie.has( menu, 'bt-menu-open' ) ) {
          resetMenu();
        }
        else {
          classie.remove( menu, 'bt-menu-close' );
          classie.add( menu, 'bt-menu-open' );
          overlay.addEventListener( eventtype, closeClickFn );
        }
      });

      if( triggerPlay ) {
        triggerPlay.addEventListener( eventtype, function( ev ) {
          ev.stopPropagation();
          ev.preventDefault();

          classie.remove( menu, 'bt-menu-close' );
          classie.add( menu, 'bt-menu-open' );
          overlay.addEventListener( eventtype, closeClickFn );
        });
      }

    }

    init();

})();


js/site.js
(function(){


  /*
        jQuery Setup
  ************************************************************************/
  jQuery.ajaxSetup({
    cache: true
  });

  function colorToHex(color) {
      if (color.substr(0, 1) === '#') {
          return color;
      }
      var digits = /(.*?)rgb\((\d+), (\d+), (\d+)\)/.exec(color);

      var red = parseInt(digits[2]);
      var green = parseInt(digits[3]);
      var blue = parseInt(digits[4]);
      var rgb = blue | (green << 8) | (red << 16);
      rgb = rgb.toString(16);
      if (rgb.length<6){
        return digits[1] + '#00' + rgb;
      }
      return digits[1] + '#' + rgb;
  }

  // Init Selectize
  $(function() {
    var options = {
      onDropdownOpen: function () {
          $(".selectize-dropdown").hide().slideToggle(250,'easeOutQuad');
      },
      onDropdownClose: function () {
          $(".selectize-dropdown").show().slideToggle(250,'easeInQuad');
      }
    };

    $('select').selectize(options);
  });

  // Init Peakaboo
  $(function() {

    var $peakaboo = $("header.peakaboo"),
        offset = $peakaboo.offset().top,
        $mini = $peakaboo.clone().addClass('mini');

        var $h1 = $mini.find('h1');
        $h1.after($("<span/>").attr({'class':$h1.attr('class')}).html($h1.html()));
        $h1.remove();

        $mini.insertAfter($peakaboo);

    var scrollTimer = null;
    $(window).scroll(function () {
        handleScroll();
    });

   function handleScroll(){
      var scrollBP = offset + $peakaboo.height()*0.75;
      var scrollTop = $(this).scrollTop();
      if (scrollTop > scrollBP){
        $mini.css({'display':'block'});
        $peakaboo.addClass('p-hidden');
      }
      else{
        $mini.css({'display':'none'});
        $peakaboo.removeClass('p-hidden');
      }

      if (scrollTop > (scrollBP + $peakaboo.height()) ){
        $mini.addClass('shadow');
      }
      else{
        $mini.removeClass('shadow');
      }
    }
  });

  $(function(){
    $(document).ready(function(){
      var url = window.location.pathname,
          $logo = $("a.logo");
      if($logo.attr('href') == url){
        $logo.on('click',function(evt){
          evt.preventDefault();
          $("html, body").animate({scrollTop:0},500, 'easeInQuad');
        });
      }
    });

    //Material Ripples
    $.material.ripples();

  });

  //Dropcap.js init
  function initDropcap(){
    var $first_p = $(".post .content>.inner>p:first-of-type");
    $first_p.each(function(){
      var $p = $(this),
          content = $p.html(),
          first_letter = content.slice(0,1),
          rest = content.slice(1);

      var drop = $('<span/>').attr({class:'dropcap'}).html(first_letter);

      $p.html('').append(drop, rest);

    });
    var dropcaps = document.querySelectorAll(".dropcap");
    window.Dropcap.layout(dropcaps, 3);
  }

  //Init expandable content
  (function() {
    var docElem = window.document.documentElement, didScroll, scrollPosition;

    // trick to prevent scrolling when opening/closing button
    function noScrollFn() {
      window.scrollTo( scrollPosition ? scrollPosition.x : 0, scrollPosition ? scrollPosition.y : 0 );
    }

    function noScroll() {
      window.removeEventListener( 'scroll', scrollHandler );
      window.addEventListener( 'scroll', noScrollFn );
    }

    function scrollFn() {
      window.addEventListener( 'scroll', scrollHandler );
    }

    function canScroll() {
      window.removeEventListener( 'scroll', noScrollFn );
      scrollFn();
    }

    function scrollHandler() {
      if( !didScroll ) {
        didScroll = true;
        setTimeout( function() { scrollPage(); }, 60 );
      }
    }

    function scrollPage() {
      scrollPosition = { x : window.pageXOffset || docElem.scrollLeft, y : window.pageYOffset || docElem.scrollTop };
      didScroll = false;
    }

    scrollFn();

    var $buttons = $( '.morph-button' );

    $buttons.each(function(){
      var el = this;

      new UIMorphingButton( el, {
        closeEl : '.icon-close',
        onBeforeOpen : function() {
          // don't allow to scroll
          noScroll();
          // hide header if it's in the way
          var $header = $("header.peakaboo.mini");
          if($(el).offset().top <= ($header.offset().top+$header.height())){
            $header.animate({"top":"-100px"},150,'easeInQuad');
          }

          // var content = $(el).find('.current .inner>.data').data('content'),
          //     $inner = $(el).find('.current section .content>.inner');
          // $inner.html(content);

          // add color to close button
          var parent_color_class = $(el).parent('.post').attr('class').replace('post','').trim();
          $(".btn-close").addClass(parent_color_class);

        },
        onAfterOpen : function() {
          // can scroll again
          canScroll();
          // add class "noscroll" to body
          classie.addClass( document.body, 'noscroll' );
          // add scroll class to main el
          classie.addClass( el, 'scroll' );
          var id = $(el).attr('data-id');
          if(id){
            window.location.hash = '/'+id;
          }

          //ajax content
          $(el).find(".content-style-overlay").removeClass("loaded");

          var self = this;
          var id = $(el).attr('data-id');

          self.loadbar.go( 10 );

          if(!self.postCache[id]){
            $.get('/blog/'+ id, function(data){
              self.postCache[id] = data;

              var html = $.parseHTML(data),
              $current = $(html).find("article.current").html(),
              $next = $(html).find("article.next").html();

              $(".page.current").html($current);
              $(".page.next").html($next);
              initDropcap();
              $(el).find(".content-style-overlay").addClass("loaded");
              self.loadbar.go( 100 );
            });
          }
          else{
            var cached = $.parseHTML(self.postCache[id]),
                $cached_current = $(cached).find("article.current"),
                $cached_next = $(cached).find("article.next");
            $(".page.current").html($cached_current);
            $(".page.next").html($cached_next);
            initDropcap();
            $(el).find(".content-style-overlay").addClass("loaded");
            self.loadbar.go( 100 );
          }

          //show close button
          $(".btn-close").removeClass('hide');
        },
        onBeforeClose : function() {
          // remove class "noscroll" to body
          classie.removeClass( document.body, 'noscroll' );
          // remove scroll class from main el
          classie.removeClass( el, 'scroll' );
          // don't allow to scroll
          noScroll();

          //hide close button
          $(".btn-close").addClass('hide');
        },
        onAfterClose : function() {
          // can scroll again
          canScroll();
          $("header.peakaboo.mini").animate({"top":"0"},150,'easeOutQuad');
          window.location.hash = '/';

          $(".page.current").html('');
          $(".page.next").html('');

          //remove color class from close button
          var parent_class = $(el).parent('.post').attr('class');
          $(".btn-close").removeClass(parent_class);

        }
      });
    });

    $(window).on('popstate', function(e){

      if( !history.state || self.initialLoad ){
        self.initialLoad = false;
        return;
      }
      var hash = window.location.hash;
      hash = hash.replace('#/','');

      if(hash!==''){
        $(".morph-button-overlay.active .icon-close").click();
      }

    });

    $(document).ready(function(){
      var hash = window.location.hash;
      hash = hash.replace('#/','');
      if(hash!==''){
        $selected = $(".morph-button[data-id="+hash+"] button");
        $selected.click();
      }
      else{
        window.location.hash = '/';
      }

      $(".btn-close").on('click',function(evt){
        evt.preventDefault();
        $(".morph-button-overlay.active .icon-close").click();
      });

      var options = {
        bg: '#5CE424',

        id: 'mynano'
      };

      var loadbar = new Nanobar( options );

      $(document).on('click','.next header',function(){
        var $this = $(this);
        var id = $this.attr('data-id');

        var $current = $this.parent().siblings(".current");

        loadbar.go( 10 );

        $current.find("header, section").css({"transform-origin":"50% 0"}).animate({"opacity": "0.2","transform":"translateY(100%) scaleX(0.5) scaleY(0.5)"},800);
        $this.css({"height":"100vh"});

        $this.parents(".content-style-overlay").removeClass("loaded");

        $current.slideUp(500,function(){
          var $self = $(this);

          $.get('/blog/'+ id, function(data){

            var html = $.parseHTML(data),
            $current = $(html).find("article.current").html(),
            $next = $(html).find("article.next").html();
            $this.css("height",'');
            $self.show();
            $(".page.current").html($current);
            $(".page.next").html($next);
            $(".content-style-overlay").addClass("loaded");
            initDropcap();
            loadbar.go( 100 );

            var stateObj = { post: id };
            history.pushState(stateObj, "", "#/"+id);
          });
        });
      });

    });
  })();

})(this);


Gruntfile.js
module.exports = function(grunt) {

  // Project configuration.
  grunt.initConfig({
    pkg: grunt.file.readJSON('package.json'),

    copy: {
      main: {
        files: [
        { expand: true, cwd: '/source/css/', src: '*.css', dest: './css/' },
        { expand: true, cwd: './source/css/', src: '*.map', dest: './css/' },
        { expand: true, cwd: './source/images/', src: ['*.png', '*.jpg', '*.gif', '*.jpeg', '*.ico'], dest: './images/' },
        { expand: true, cwd: './bower_components/bourbon/app/assets/stylesheets/', src: '**', dest: './source/css/bourbon/'},
        { expand: true, cwd: './bower_components/selectize/dist/js/standalone/', src: '*.min.js', dest: './js/selectize/'},
        { expand: true, cwd: './bower_components/nanobar/', src: '*.min.js', dest: './js/nanobar/'},
        { expand: true, cwd: './bower_components/modernizr/', src: 'modernizr.js', dest: './js/'},
        { expand: true, cwd: './node_modules/dropcap.js/', src: 'dropcap.min.js', dest: './js/'}

        ]
      }
    },

    watch: {
      all: {
        options: {
          livereload: 9002
        },
        files: ['index.html','source/css/**/*.scss'],
        tasks: ['default']
      }
    },

    sass: {
      build: {
        options: {
          style: 'expanded',
          precision: 8
        },
        files: {
          './css/style.css': './source/css/style.scss'
        }
      }
    },

    uglify: {
      js: {
        files: { 'scripts/site.min.js': ['js/jquery.min.js','js/easing.js','js/classie.js','js/menu.js','js/material/*.js','js/selectize/*.js','js/nanobar/*.js', 'js/transform.js','js/uiMorphingButton.js','js/dropcap.min.js','js/404.js','js/site.js'] },
        options: {
            preserveComments: false,
            sourceMap: true
        }
      }
    },

    cssmin: {
      options: {
        shorthandCompacting: false,
        roundingPrecision: -1,
        sourceMap:true
      },
      target: {
        files: {
          'styles/site.min.css': ['css/ripples.min.css', 'css/font-awesome.min.css','css/material-icons.css','css/component.css','css/style.css']
        }
      }
    },

    zip: {
      main: {
        src: ['_layouts/**','blog/*','data/**','fonts/*','styles/*','scripts/*','index.html', '404.html.liquid'],
        dest: 'theme.zip'
      }
    },

    nodeunit: {
      all: ['test/*_tests.js']
    }
  });

  // load all grunt tasks
  grunt.loadNpmTasks('grunt-contrib-copy');
  grunt.loadNpmTasks('grunt-contrib-cssmin');
  grunt.loadNpmTasks('grunt-contrib-sass');
  grunt.loadNpmTasks('grunt-contrib-uglify');
  grunt.loadNpmTasks('grunt-contrib-watch');
  grunt.loadNpmTasks('grunt-zip');



  //if you choose to use scss, or any preprocessor, you can add it here
  grunt.registerTask('default', ['sass', 'copy']);

  grunt.registerTask('minify', ['sass', 'copy', 'uglify', 'cssmin']);

  grunt.registerTask('active-watch', ['sass', 'copy', 'uglify', 'cssmin', 'watch']);

  grunt.registerTask('active-watch-minify', ['sass', 'copy', 'watch']);

  //travis CI task
  grunt.registerTask('travis', ['sass', 'copy', 'uglify', 'cssmin', 'watch', 'zip', 'nodeunit']);

};


js/classie.js
/*!
* classie - class helper functions
* from bonzo https://github.com/ded/bonzo
* 
* classie.has( elem, 'my-class' ) -> true/false
* classie.add( elem, 'my-new-class' )
* classie.remove( elem, 'my-unwanted-class' )
* classie.toggle( elem, 'my-class' )
*/

/*jshint browser: true, strict: true, undef: true */

( function( window ) {

'use strict';

// class helper functions from bonzo https://github.com/ded/bonzo

function classReg( className ) {
return new RegExp("(^|\\s+)" + className + "(\\s+|$)");
}

// classList support for class management
// altho to be fair, the api sucks because it won't accept multiple classes at once
var hasClass, addClass, removeClass;

if ( 'classList' in document.documentElement ) {
hasClass = function( elem, c ) {
  return elem.classList.contains( c );
};
addClass = function( elem, c ) {
  elem.classList.add( c );
};
removeClass = function( elem, c ) {
  elem.classList.remove( c );
};
}
else {
hasClass = function( elem, c ) {
  return classReg( c ).test( elem.className );
};
addClass = function( elem, c ) {
  if ( !hasClass( elem, c ) ) {
    elem.className = elem.className + ' ' + c;
  }
};
removeClass = function( elem, c ) {
  elem.className = elem.className.replace( classReg( c ), ' ' );
};
}

function toggleClass( elem, c ) {
var fn = hasClass( elem, c ) ? removeClass : addClass;
fn( elem, c );
}

window.classie = {
// full names
hasClass: hasClass,
addClass: addClass,
removeClass: removeClass,
toggleClass: toggleClass,
// short names
has: hasClass,
add: addClass,
remove: removeClass,
toggle: toggleClass
};

})( window );


js/404.js
//jQTouch
(function($){$.jQTouch=function(_2){$.support.WebKitCSSMatrix=(typeof WebKitCSSMatrix=="object");$.support.touch=(typeof Touch=="object");$.support.WebKitAnimationEvent=(typeof WebKitTransitionEvent=="object");var _3,$head=$("head"),hist=[],newPageCount=0,jQTSettings={},hashCheck,currentPage,orientation,isMobileWebKit=RegExp(" Mobile/").test(navigator.userAgent),tapReady=true,lastAnimationTime=0,touchSelectors=[],publicObj={},extensions=$.jQTouch.prototype.extensions,defaultAnimations=["slide","flip","slideup","swap","cube","pop","dissolve","fade","back"],animations=[],hairextensions="";init(_2);function init(_4){var _5={addGlossToIcon:true,backSelector:".back, .cancel, .goback",cacheGetRequests:true,cubeSelector:".cube",dissolveSelector:".dissolve",fadeSelector:".fade",fixedViewport:true,flipSelector:".flip",formSelector:"form",fullScreen:true,fullScreenClass:"fullscreen",icon:null,touchSelector:"a, .touch",popSelector:".pop",preloadImages:false,slideSelector:"body > * > ul li a",slideupSelector:".slideup",startupScreen:null,statusBar:"default",submitSelector:".submit",swapSelector:".swap",useAnimations:true,useFastTouch:true};jQTSettings=$.extend({},_5,_4);if(jQTSettings.preloadImages){for(var i=jQTSettings.preloadImages.length-1;i>=0;i--){(new Image()).src=jQTSettings.preloadImages[i];}}if(jQTSettings.icon){var _7=(jQTSettings.addGlossToIcon)?"":"-precomposed";hairextensions+="<link rel=\"apple-touch-icon"+_7+"\" href=\""+jQTSettings.icon+"\" />";}if(jQTSettings.startupScreen){hairextensions+="<link rel=\"apple-touch-startup-image\" href=\""+jQTSettings.startupScreen+"\" />";}if(jQTSettings.fixedViewport){hairextensions+="<meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0;\"/>";}if(jQTSettings.fullScreen){hairextensions+="<meta name=\"apple-mobile-web-app-capable\" content=\"yes\" />";if(jQTSettings.statusBar){hairextensions+="<meta name=\"apple-mobile-web-app-status-bar-style\" content=\""+jQTSettings.statusBar+"\" />";}}if(hairextensions){$head.append(hairextensions);}$(document).ready(function(){for(var i in extensions){var fn=extensions[i];if($.isFunction(fn)){$.extend(publicObj,fn(publicObj));}}for(var i in defaultAnimations){var _a=defaultAnimations[i];var _b=jQTSettings[_a+"Selector"];if(typeof (_b)=="string"){addAnimation({name:_a,selector:_b});}}touchSelectors.push("input");touchSelectors.push(jQTSettings.touchSelector);touchSelectors.push(jQTSettings.backSelector);touchSelectors.push(jQTSettings.submitSelector);$(touchSelectors.join(", ")).css("-webkit-touch-callout","none");$(jQTSettings.backSelector).tap(liveTap);$(jQTSettings.submitSelector).tap(submitParentForm);_3=$("body");if(jQTSettings.fullScreenClass&&window.navigator.standalone==true){_3.addClass(jQTSettings.fullScreenClass+" "+jQTSettings.statusBar);}_3.bind("touchstart",handleTouch).bind("orientationchange",updateOrientation).trigger("orientationchange").submit(submitForm);if(jQTSettings.useFastTouch&&$.support.touch){_3.click(function(e){var _d=$(e.target);if(_d.attr("target")=="_blank"||_d.attr("rel")=="external"||_d.is("input[type=\"checkbox\"]")){return true;}else{return false;}});_3.mousedown(function(e){var _f=(new Date()).getTime()-lastAnimationTime;if(_f<200){return false;}});}if($("body > .current").length==0){currentPage=$("body > *:first");}else{currentPage=$("body > .current:first");$("body > .current").removeClass("current");}$(currentPage).addClass("current");location.hash=$(currentPage).attr("id");addPageToHistory(currentPage);scrollTo(0,0);dumbLoopStart();});}function goBack(to){if(hist.length>1){var _11=Math.min(parseInt(to||1,10),hist.length-1);if(isNaN(_11)&&typeof (to)==="string"&&to!="#"){for(var i=1,length=hist.length;i<length;i++){if("#"+hist[i].id===to){_11=i;break;}}}if(isNaN(_11)||_11<1){_11=1;}var _13=hist[0].animation;var _14=hist[0].page;hist.splice(0,_11);var _15=hist[0].page;animatePages(_14,_15,_13,true);return publicObj;}else{console.error("No pages in history.");return false;}}function goTo(_16,_17){var _18=hist[0].page;if(typeof (_16)==="string"){_16=$(_16);}if(typeof (_17)==="string"){for(var i=animations.length-1;i>=0;i--){if(animations[i].name===_17){_17=animations[i];break;}}}if(animatePages(_18,_16,_17)){addPageToHistory(_16,_17);return publicObj;}else{console.error("Could not animate pages.");return false;}}function getOrientation(){return orientation;}function liveTap(e){var $el=$(e.target);if($el.attr("nodeName")!=="A"){$el=$el.parent("a");}var _1c=$el.attr("target"),hash=$el.attr("hash"),animation=null;if(tapReady==false||!$el.length){console.warn("Not able to tap element.");return false;}if($el.attr("target")=="_blank"||$el.attr("rel")=="external"){return true;}for(var i=animations.length-1;i>=0;i--){if($el.is(animations[i].selector)){animation=animations[i];break;}}if(_1c=="_webapp"){window.location=$el.attr("href");}else{if($el.is(jQTSettings.backSelector)){goBack(hash);}else{if(hash&&hash!="#"){$el.addClass("active");goTo($(hash).data("referrer",$el),animation);}else{$el.addClass("loading active");showPageByHref($el.attr("href"),{animation:animation,callback:function(){$el.removeClass("loading");setTimeout($.fn.unselect,250,$el);},$referrer:$el});}}}return false;}function addPageToHistory(_1e,_1f){var _20=_1e.attr("id");hist.unshift({page:_1e,animation:_1f,id:_20});}function animatePages(_21,_22,_23,_24){if(_22.length===0){$.fn.unselect();console.error("Target element is missing.");return false;}$(":focus").blur();scrollTo(0,0);var _25=function(_26){if(_23){_22.removeClass("in reverse "+_23.name);_21.removeClass("current out reverse "+_23.name);}else{_21.removeClass("current");}_22.trigger("pageAnimationEnd",{direction:"in"});_21.trigger("pageAnimationEnd",{direction:"out"});clearInterval(dumbLoop);currentPage=_22;location.hash=currentPage.attr("id");dumbLoopStart();var _27=_22.data("referrer");if(_27){_27.unselect();}lastAnimationTime=(new Date()).getTime();tapReady=true;};_21.trigger("pageAnimationStart",{direction:"out"});_22.trigger("pageAnimationStart",{direction:"in"});if($.support.WebKitAnimationEvent&&_23&&jQTSettings.useAnimations){_22.one("webkitAnimationEnd",_25);tapReady=false;_22.addClass(_23.name+" in current "+(_24?" reverse":""));_21.addClass(_23.name+" out"+(_24?" reverse":""));}else{_22.addClass("current");_25();}return true;}function dumbLoopStart(){dumbLoop=setInterval(function(){var _28=currentPage.attr("id");if(location.hash==""){location.hash="#"+_28;}else{if(location.hash!="#"+_28){try{goBack(location.hash);}catch(e){console.error("Unknown hash change.");}}}},100);}function insertPages(_29,_2a){var _2b=null;$(_29).each(function(_2c,_2d){var _2e=$(this);if(!_2e.attr("id")){_2e.attr("id","page-"+(++newPageCount));}_2e.appendTo(_3);if(_2e.hasClass("current")||!_2b){_2b=_2e;}});if(_2b!==null){goTo(_2b,_2a);return _2b;}else{return false;}}function showPageByHref(_2f,_30){var _31={data:null,method:"GET",animation:null,callback:null,$referrer:null};var _32=$.extend({},_31,_30);if(_2f!="#"){$.ajax({url:_2f,data:_32.data,type:_32.method,success:function(_33,_34){var _35=insertPages(_33,_32.animation);if(_35){if(_32.method=="GET"&&jQTSettings.cacheGetRequests&&_32.$referrer){_32.$referrer.attr("href","#"+_35.attr("id"));}if(_32.callback){_32.callback(true);}}},error:function(_36){if(_32.$referrer){_32.$referrer.unselect();}if(_32.callback){_32.callback(false);}}});}else{if($referrer){$referrer.unselect();}}}function submitForm(e,_38){var _39=(typeof (e)==="string")?$(e):$(e.target);if(_39.length&&_39.is(jQTSettings.formSelector)&&_39.attr("action")){showPageByHref(_39.attr("action"),{data:_39.serialize(),method:_39.attr("method")||"POST",animation:animations[0]||null,callback:_38});return false;}return true;}function submitParentForm(e){var _3b=$(this).closest("form");if(_3b.length){evt=jQuery.Event("submit");evt.preventDefault();_3b.trigger(evt);return false;}return true;}function addAnimation(_3c){if(typeof (_3c.selector)=="string"&&typeof (_3c.name)=="string"){animations.push(_3c);$(_3c.selector).tap(liveTap);touchSelectors.push(_3c.selector);}}function updateOrientation(){orientation=window.innerWidth<window.innerHeight?"profile":"landscape";_3.removeClass("profile landscape").addClass(orientation).trigger("turn",{orientation:orientation});}function handleTouch(e){var $el=$(e.target);if(!$(e.target).is(touchSelectors.join(", "))){var _3f=$(e.target).closest("a");if(_3f.length){$el=_3f;}else{return;}}if(event){var _40=null,startX=event.changedTouches[0].clientX,startY=event.changedTouches[0].clientY,startTime=(new Date).getTime(),deltaX=0,deltaY=0,deltaT=0;$el.bind("touchmove",touchmove).bind("touchend",touchend);_40=setTimeout(function(){$el.makeActive();},100);}function touchmove(e){updateChanges();var _42=Math.abs(deltaX);var _43=Math.abs(deltaY);if(_42>_43&&(_42>35)&&deltaT<1000){$el.trigger("swipe",{direction:(deltaX<0)?"left":"right"}).unbind("touchmove touchend");}else{if(_43>1){$el.removeClass("active");}}clearTimeout(_40);}function touchend(){updateChanges();if(deltaY===0&&deltaX===0){$el.makeActive();$el.trigger("tap");}else{$el.removeClass("active");}$el.unbind("touchmove touchend");clearTimeout(_40);}function updateChanges(){var _44=event.changedTouches[0]||null;deltaX=_44.pageX-startX;deltaY=_44.pageY-startY;deltaT=(new Date).getTime()-startTime;}}$.fn.unselect=function(obj){if(obj){obj.removeClass("active");}else{$(".active").removeClass("active");}};$.fn.makeActive=function(){return $(this).addClass("active");};$.fn.swipe=function(fn){if($.isFunction(fn)){return this.each(function(i,el){$(el).bind("swipe",fn);});}};$.fn.tap=function(fn){if($.isFunction(fn)){var _4a=(jQTSettings.useFastTouch&&$.support.touch)?"tap":"click";return $(this).live(_4a,fn);}else{$(this).trigger("tap");}};publicObj={getOrientation:getOrientation,goBack:goBack,goTo:goTo,addAnimation:addAnimation,submitForm:submitForm};return publicObj;};$.jQTouch.prototype.extensions=[];$.jQTouch.addExtension=function(_4b){$.jQTouch.prototype.extensions.push(_4b);};})(jQuery);

var squareBoxes = function(){
  $(function(){

    var sectionW = $('.draw_canvas').width();

    var boxW = sectionW/20;

    var rows = 40;

    var inc = Math.round((sectionW/boxW))*rows;


    for(var x=1; x<inc+1; x++){

      var code='<div class="card_contianer" id="' + x +'"><div class="card"><div class="front face"></div><div class="back face center"></div></div></div>';

      $('.draw_canvas').append(code);

    }


  //Button Events
     document.onkeydown = function(evt) {

        evt = evt || window.event;

        switch (evt.keyCode) {
          case 65:
            evt.preventDefault();
            $('#draw').toggleClass('active');
            break;
          case 68:
            evt.preventDefault();
            $('#mo_to_draw').toggleClass('active');
            break;
        }
      };


    $('#draw, #mo_to_draw').on('click',function(){

      $(this).toggleClass('active');

    });

    $('.card_contianer').bind('mouseenter',function(){

        if($('#draw').hasClass('active') && $('#mo_to_draw').hasClass('active') ){

          $(this).toggleClass('zoom_effect');

        }

         var hue = 'rgb('
          + (Math.floor(Math.random() * 256)) + ','
          + (Math.floor(Math.random() * 256)) + ','
          + (Math.floor(Math.random() * 256)) + ')';

        $(this).find('.face.back').css({'background-color': hue});

    });


    $('.card_contianer').bind('click',function(){

      if($('#draw').hasClass('active')){

        $(this).toggleClass('zoom_effect');

      }
         var hue = 'rgb('
          + (Math.floor(Math.random() * 256)) + ','
          + (Math.floor(Math.random() * 256)) + ','
          + (Math.floor(Math.random() * 256)) + ')';

        $(this).find('.face.back').css({'background-color': hue});
    });


    $("#clear").bind('click', function(){


      $('#draw').addClass('active');

      $('.zoom_effect').toggleClass('zoom_effect');

    });

    $('#print').bind('click', function(){

      var get_clicked = $('.zoom_effect').length;

      var list1=$('.zoom_effect');

      var list2=[];

      for(var i=0; i<get_clicked; i++){


        var id =$(list1[i]).attr('id');

        list2.push(id);

      }

      var print_list = '"' + list2.join('","') + '"';

      console.log(print_list);

    });

  });

  //setup for drawing
    function draw(){

      $('#draw, #mo_to_draw').addClass('active');

      $('#click_to_draw').click();

    }//end draw

    $(document).ready(function(){
      draw();
      $('#clear').click();
    });
};
